### PR TITLE
feat(Ortganizations): add organizations resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:

1. add organizations resource
2. add organizations data source
3. add organizations unit resource

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add organizations resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
go test ./huaweicloud/services/acceptance/organizations -v -run TestAccOrganizations_basic -timeout 360m
=== RUN   TestAccOrganizations_basic
=== PAUSE TestAccOrganizations_basic
=== CONT  TestAccOrganizations_basic
--- PASS: TestAccOrganizations_basic (943.36s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/organizations     943.419s

go test ./huaweicloud/services/acceptance/organizations -v -run TestAccDatasourceOrganizations_basic -timeout 360m
=== RUN   TestAccDatasourceOrganizations_basic
=== PAUSE TestAccDatasourceOrganizations_basic
=== CONT  TestAccDatasourceOrganizations_basic
--- PASS: TestAccDatasourceOrganizations_basic (504.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/organizations     504.992s

go test ./huaweicloud/services/acceptance/organizations -v -run TestAccOrganizationsUnit_basic -timeout 360m
=== RUN   TestAccOrganizationsUnit_basic
=== PAUSE TestAccOrganizationsUnit_basic
=== CONT  TestAccOrganizationsUnit_basic
--- PASS: TestAccOrganizationsUnit_basic (943.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/organizations     943.857s
```

### CODE REVIEW

File: docs/data-sources/organizations.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+* `created_at` - Indicates the time when the organization was created.
+
+* `root_id` - Indicates the ID of the root.
+
+* `root_name` - Indicates the name of the root.
+
+* `root_urn` - Indicates the urn of the root.
+
+* `root_tags` - Indicates the key/value to attach to the root.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/organizations.md]: this is an attribute, so how about this
```suggestion
* `root_tags` - Indicates the key/value attached to the root.
```

File: docs/resources/organizations.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `root_tags` - (Optional, Map) Specifies the key/value to attach to the root.
```
Comment (Jason-Zhang9309) [inline docs/resources/organizations.md]: what the difference between this and common tags?

File: huaweicloud/services/organizations/resource_huaweicloud_organizations.go [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+	// updateOrganizations: update Organizations
+	var (
+		updateOrganizationsProduct = "organizations"
+	)
+	updateOrganizationsClient, err := cfg.NewServiceClient(updateOrganizationsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating Organizations Client: %s", err)
+	}
+
+	if d.HasChange("root_tags") {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/organizations/resource_huaweicloud_organizations.go]: this line is unnecessary

File: huaweicloud/services/organizations/resource_huaweicloud_organizations_unit.go [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+	updateOrganizationsUnitClient, err := cfg.NewServiceClient(updateOrganizationsUnitProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating Organizations Client: %s", err)
+	}
+
+	updateOrganizationsUnitHasChanges := []string{
+		"name",
+	}
+
+	if d.HasChanges(updateOrganizationsUnitHasChanges...) {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/organizations/resource_huaweicloud_organizations_unit.go]:
```suggestion
	if d.HasChange("name") {
```

File: huaweicloud/services/organizations/resource_huaweicloud_organizations.go [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+	getTagsOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+
+	getTagsResp, err := client.Request("GET", getTagsPath, &getTagsOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error get Organizations tags:: %s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/organizations/resource_huaweicloud_organizations.go]:
```suggestion
		return nil, fmt.Errorf("error get Organizations tags: %s", err)
```

File: huaweicloud/services/acceptance/organizations/resource_huaweicloud_organizations_test.go [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+func testOrganizations_basic_update() string {
+	return `
+resource "huaweicloud_organizations" "test" {
+  root_tags = {
+    "key3" = "value3"
+    "key4" = "value4"
+  }
+}
+`
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/organizations/resource_huaweicloud_organizations_test.go]: no refer variables, constant just ok

File: huaweicloud/config/endpoints.go [hunk 7]
```diff
@@ -766,6 +766,12 @@ var allServiceCatalog = map[string]ServiceCatalog{
 		WithOutProjectID: true,
 		Product:          "NAT",
 	},
+	"organizations": {
+		Name:             "organizations",
+		Version:          "v1",
+		WithOutProjectID: true,
```
Comment (ShiChangkuo) [inline huaweicloud/config/endpoints.go]: according to https://developer.huaweicloud.com/endpoint?Organizations, the endpoint of Organizations is  global and without region, so we should  add ```Scope: "global",```

File: huaweicloud/config/endpoints.go [hunk 8]
```diff
@@ -766,6 +766,13 @@ var allServiceCatalog = map[string]ServiceCatalog{
 		WithOutProjectID: true,
 		Product:          "NAT",
 	},
+	"organizations": {
```
Comment (ShiChangkuo) [inline huaweicloud/config/endpoints.go]: it's better to add this item after **rms**.

File: docs/data-sources/organizations.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+## Example Usage
+
+```hcl
+data "huaweicloud_organizations" "test"{
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
```
Comment (ShiChangkuo) [inline docs/data-sources/organizations.md]: the service is global, so `region` is useless.

```
There are no arguments available for this data source.
```

File: docs/resources/organizations_unit.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+$ terraform import huaweicloud_organizations_unit.test <id>
+```
+
+Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
+API response, security or some other reason. The missing attributes include: `parent_id`. It is generally recommended
+running `terraform plan` after importing the organizations unit. You can then decide if changes should be applied to
+the organizations unit, or the resource definition should be updated to align with the organizations unit. Also you
+can ignore changes as below.
+
+```
```
Comment (ShiChangkuo) [inline docs/resources/organizations_unit.md]:
```suggestion
```hcl
```

File: huaweicloud/services/organizations/data_source_huaweicloud_organizations.go [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+	if diagErr != nil {
+		return diagErr
+	}
+
+	getOrganizationsRootRespBody, diagErr := getOrganizationsRoot(d, getOrganizationsClient)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	dataSourceId, err := uuid.GenerateUUID()
```
Comment (ShiChangkuo) [inline huaweicloud/services/organizations/data_source_huaweicloud_organizations.go]: please use **organization.id** as the data source ID

File: huaweicloud/services/organizations/data_source_huaweicloud_organizations_organization.go [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+	var (
+		getOrganizationProduct = "organizations"
+	)
+	getOrganizationClient, err := cfg.NewServiceClient(getOrganizationProduct, "")
+	if err != nil {
+		return diag.Errorf("error creating Organizations Client: %s", err)
+	}
+
+	getOrganizationRespBody, diagErr := getOrganization(d, getOrganizationClient)
+	if diagErr != nil {
```
Comment (ShiChangkuo) [inline huaweicloud/services/organizations/data_source_huaweicloud_organizations_organization.go]: For data source, we should return error even if we got 404 status code

File: huaweicloud/services/acceptance/acceptance.go [hunk 13]
```diff
@@ -184,6 +184,14 @@ func preCheckRequiredEnvVars(t *testing.T) {
 	}
 }
 
+// lintignore:AT003
+func TestAccPreCheckDeprecatedEnvironment(t *testing.T) {
+	// Do not run the test if this is a deprecated testing environment.
+	if HW_DEPRECATED_ENVIRONMENT != "" {
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/acceptance.go]: please use a new environment for organizations service acceptance test.

File: huaweicloud/services/acceptance/organizations/resource_huaweicloud_organizations_organizational_unit_test.go [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+					resource.TestCheckResourceAttr(rName, "name", updateName),
+					resource.TestCheckResourceAttr(rName, "tags.key3", "value3"),
+					resource.TestCheckResourceAttr(rName, "tags.key4", "value4"),
+				),
+			},
+			{
+				ResourceName:            rName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parent_id"},
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/organizations/resource_huaweicloud_organizations_organizational_unit_test.go:89]: can we get the parent id from urn?

**Pull request merged**